include/gul14/variant.h

```diff
@@ -62,10 +62,84 @@ class bad_variant_access : public std::exception
  * only the stored type is accessed. The implementation in the library is a backport from
  * C++17 and should behave like
  * [std::variant](https://en.cppreference.com/w/cpp/utility/variant).
+ *
+ * \since GUL version 2.9.0
+ *
+ * \see visit(), make_overload_set()
  */
 template <typename... Ts>
 class variant;
 
+/**
+ * Call a visitor function with the actual objects stored in the given variants.
+ *
+ * This function works like [std::visit](https://en.cppreference.com/w/cpp/utility/variant/visit)
+ * from the C++17 standard library. It is often convenient to use it with
+ * \ref gul14::make_overload_set "make_overload_set()":
+ * \code{.cpp}
+ * gul14::variant<int, double> v = 1.0;
+ * gul14::visit(gul14::make_overload_set(
+ *     [](int) { std::cout << "int\n"; },
+ *     [](double) { std::cout << "double\n"; }
+ * ), v);
+ * \endcode
+ *
+ * \since GUL version 2.9.0
+ */
+template <typename Visitor, typename... Variants>
+inline constexpr decltype(auto) visit(Visitor&& visitor_fct, Variants&&... variants);
+
+/**
+ * A function object that works like an overload set of functions.
+ *
+ * This variadic class template inherits from an arbitrary number of function objects
+ * (typically lambdas) and imports their definitions of `operator()` into an overload set
+ * for its own `operator()`. This is primarily helpful for use with \ref gul14::visit
+ * "visit()". Constructing an OverloadSet object manually can be cumbersome. The function
+ * \ref gul14::make_overload_set "make_overload_set()" can help with that.
+ *
+ * \since GUL version 2.9.2
+ */
+template <typename Fct1, typename... Fcts>
+struct OverloadSet : Fct1, OverloadSet<Fcts...>
+{
+    using Fct1::operator();
+    using OverloadSet<Fcts...>::operator();
+
+    OverloadSet(Fct1 f1, Fcts... fs)
+        : Fct1(f1), OverloadSet<Fcts...>(fs...)
+    {}
+};
+
+/// OverloadSet for a single function object.
```

Comment:
I wonder if the single function object should be declared before the variadic version.

Reply:
I think it does not matter. Won't change until a compiler complains.